share/mediation_integrate.F

```diff
@@ -2433,3 +2447,65 @@ END SUBROUTINE med_read_wrf_chem_emissopt3
 #endif
 
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+SUBROUTINE med_read_qna_emissions ( grid , config_flags )
+  ! Driver layer
+   USE module_domain    , ONLY : domain , domain_clock_get
+   USE module_io_domain
+   USE module_timing
+   USE module_configure , ONLY : grid_config_rec_type
+  ! Model layer
+   USE module_bc_time_utilities
+#ifdef DM_PARALLEL
+   USE module_dm
+#endif
+   USE module_date_time
+   USE module_utility
+
+   IMPLICIT NONE
+
+  ! Arguments
+   TYPE(domain)                               :: grid
+
+   TYPE (grid_config_rec_type) , INTENT(IN)   :: config_flags
+
+  ! Local data
+   LOGICAL, EXTERNAL                      :: wrf_dm_on_monitor
+
+   INTEGER                                :: ierr, efid
+   REAL                                   :: time, tupdate
+   real, allocatable :: dumc0(:,:,:)
+   CHARACTER (LEN=256)                    :: message, current_date_char, date_string
+   CHARACTER (LEN=256)                    :: inpname
+
+#include "wrf_io_flags.h"
+
+      CALL domain_clock_get( grid, current_timestr=current_date_char )
+
+      CALL construct_filename1 ( inpname , 'wrfqnainp' , grid%id , 2 )
+      WRITE(message,*)'mediation_integrate: med_read_qna_emissions: Open file ',TRIM(inpname)
+      CALL wrf_message( TRIM(message) )
+
+     if( grid%auxinput17_oid .NE. 0 ) then
+       CALL close_dataset ( grid%auxinput17_oid , config_flags , "DATASET=AUXINPUT17" )
+     endif
+
+      CALL open_r_dataset ( grid%auxinput17_oid, TRIM(inpname) , grid , config_flags, &
+                              "DATASET=AUXINPUT17", ierr )
+        IF ( ierr .NE. 0 ) THEN
+           WRITE( message , * ) 'med_read_qna_emissions: error opening ', TRIM( inpname )
+           CALL wrf_error_fatal( TRIM( message ) )
+        ENDIF
+
+         WRITE(message,*)'mediation_integrate: med_read_qna_emissions: Read surface aerosol emissions at time ',&
+         TRIM(current_date_char)
+         CALL wrf_message( TRIM(message) )
+
+         CALL wrf_debug (00 , 'mediation_integrate: calling input_auxinput17' )
+         CALL input_auxinput17 ( grid%auxinput17_oid, grid , config_flags , ierr )
+
+         CALL close_dataset ( grid%auxinput17_oid , config_flags , "DATASET=AUXINPUT17" )
+
+   CALL wrf_debug (00 , 'mediation_integrate: med_read_qna_emissions: exit' )
+
+END SUBROUTINE med_read_qna_emissions
```

Comment:
@davegill I don't understand why this subroutine is needed. Is there an equivalent for wrflowinp?

Reply:
@twjuliano 
Tim,
WRF is able to use the share/input_wrf.F file to bring in data. Is there a special need for this routine?

Reply:
@weiwangncar @davegill This subroutine is used to read the wrfqnainp file. At the time, it seemed like the best way to bring in the data. If you prefer me to use share/input_wrf.F, then I can do that perhaps with some guidance. Thanks.

Reply:
@davegill Can you provide a rough outline for Tim to follow using input_wrf.F?